### ISSUE TITLE
Building Problems

### ISSUE THREAD

**Issue by aggsol** [slowriot/libtelegram#1]
* the Makefile makes nothing. How about something more generic like CMake? that can generate CodeBlock projects and more!

```
$ make
Makefile:49: warning: overriding commands for target `placeholder'
Makefile:38: warning: ignoring old commands for target `placeholder'
make: *** No rule to make target `placeholder.cbp', needed by `placeholder'.  Stop.
```

* usage of experimental headers makes the library cutting-edge but just not usable.  How about boost::optional instead?

The current state makes it really hard to use libtelegram.

**Comment by slowriot** [comment]
There are two separate issues here.

First is you're trying to use make without reading the readme - this is a header-only library, so there is nothing to compile before you can start using it.  The makefile is there to help you build the examples; you just need to specify which you want to build, with the correct target:

    make TARGET=example_echo

Your second issue is with std::experimental, and if that's a problem I suggest opening another issue, but I'll address it quickly here.  Experimental::optional has been available in stable releases of GCC for a long while now, and it's available on any system you can run `make` on.  I'd be very surprised if you don't have it available in your distribution, and it's available even in MinGW-W64 on Windows.  I believe pulling in compiled boost libraries as dependencies is a far worse solution.

**Comment by aggsol** [comment]
Oh sorry, my bad. I have an Ubuntu 14.04.1 LTS with g++ 4.8.5 I will try to upgrade then.

**Comment by slowriot** [comment]
std::experimental::optional is available from GCC 4.9: https://gcc.gnu.org/gcc-4.9/changes.html

Note however this is already a two year old version of GCC (although what you're running has been outdated for about three years).  The current stable gcc on debian-based distributions is 6.3, and I would recommend upgrading to at least 5.0 for the optimisation features and bugfixes, if nothing else.
